plugins/imagebase/plugin.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or http://ckeditor.com/license
+ */
+
+( function() {
+	'use strict';
+
+	function createWidgetDefinition( editor, options ) {
+		var defaultTemplate = new CKEDITOR.template(
+			'<figure class="{basicClass}">' +
+				'<img alt="" src="" />' +
+				'<figcaption>{captionPlaceholder}</figcaption>' +
+			'</figure>' );
+
+		function joinClasses( basicClass, additionalClasses ) {
+			var toApply = additionalClasses.slice();
+
+			toApply.unshift( basicClass );
+
+			return toApply.join( ',' );
+		}
+
+		/**
+		 * This is an abstract class that describes a definition of an image widget.
+		 * It is a return type of {@link CKEDITOR.plugins.imagebase#createWidget} method.
+		 *
+		 * Note that because the image widget is a type of a widget, this definition extends
+		 * {@link CKEDITOR.plugins.widget.definition}.
+		 * It adds several parts of image and implements the {@link CKEDITOR.plugins.widget.definition#upcast} callback.
+		 * This callback should not be overwritten.
+		 *
+		 * @abstract
+		 * @class CKEDITOR.plugins.imagebase.imageWidgetDefinition
+		 * @mixins CKEDITOR.plugins.widget.definition
+		 */
+		return {
+			/**
+			 * @inheritdoc
```

Comment:
There's no need to explicitly set `@inheritdoc` if you're not modifying docs of inherited object, it will be implicitly inherited.